internal/config/config.go

```diff
@@ -0,0 +1,121 @@
+package config
+
+import (
+	"fmt"
+	"github.com/rs/zerolog"
+	"strings"
+	"time"
+
+	clowder "github.com/redhatinsights/app-common-go/pkg/api/v1"
+	"github.com/spf13/viper"
+)
+
+type DatabaseT struct {
+	Host        string
+	Port        uint16
+	Name        string
+	User        string
+	Password    string
+	MaxIdleTime time.Duration
+	MaxLifetime time.Duration
+	MaxOpenConn int
+	MaxIdleConn int
+	LogLevel    int
+}
+
+type LoggingT struct {
+	Level  int
+	Stdout bool
+}
+type CloudwatchT struct {
+	Enabled bool
+	Group   string
+	Stream  string
+}
+type FeatureFlagsT struct {
+	Environment string
+	ExitOnPanic bool
+}
+
+type GlobalConfigT struct {
+	Database DatabaseT
+	Logging LoggingT
+	Cloudwatch CloudwatchT
+	Features FeatureFlagsT
+}
+
+var config GlobalConfigT
+var Database *DatabaseT
+var Logging *LoggingT
+var Cloudwatch *CloudwatchT
+var Features *FeatureFlagsT
+
+func init() {
+	var err error
+
+	viper.SetDefault("logging.level", 1)
+	viper.SetDefault("database.host", "localhost")
+	viper.SetDefault("database.port", 5432)
+	viper.SetDefault("database.loglevel", 1)
+	viper.SetDefault("cloudwatch.enabled", false)
```

Comment:
These default values do not match with the YAML file, what is the source of "truth" in this design? I would prefer to have just a single place where we keep all the defaults. If this is the place and YAML is not, then please comment all the lines in the YAML file so its all just comments so we indicate to the user that these are all just examples. Maybe you can drop a comment in the YAML that the effective default values are in this file.

Reply:
I've commented out the yaml file